qiskit_experiments/library/calibration/fine_drag.py

```diff
@@ -0,0 +1,271 @@
+        """
+        super().__init__([qubit])
+
+    @staticmethod
+    def _pre_circuit() -> QuantumCircuit:
+        """Return the quantum circuit to apply before repeating the Rp - Rz - Rp - Rz gates."""
+        return QuantumCircuit(1)
+
+    @staticmethod
+    def _post_circuit() -> QuantumCircuit:
+        """Return the quantum circuit to apply after repeating the Rp - Rz - Rp - Rz gates."""
+        return QuantumCircuit(1)
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Create the circuits for the fine DRAG calibration experiment.
+
+        Args:
+            backend: A backend object.
+
+        Returns:
+            A list of circuits with a variable number of gates. Each gate has the same
+            pulse schedule.
+        """
+        schedule, circuits = self.experiment_options.schedule, []
+
+        drag_gate = Gate(name=schedule.name, num_qubits=1, params=[])
+
+        for repetition in self.experiment_options.repetitions:
+            circuit = self._pre_circuit()
+
+            if self.experiment_options.add_sx:
+                circuit.sx(0)
+
+            for _ in range(repetition):
+                circuit.append(drag_gate, (0,))
+                circuit.rz(np.pi, 0)
+                circuit.append(drag_gate, (0,))
+                circuit.rz(np.pi, 0)
+
+            circuit.ry(np.pi / 2, 0)  # Maps unwanted Z rotations to qubit population.
+
+            circuit.compose(self._post_circuit(), inplace=True)
+
+            circuit.measure_all()
+            circuit.add_calibration(schedule.name, self._physical_qubits, schedule, params=[])
+
+            circuit.metadata = {
+                "experiment_type": self._type,
+                "qubits": self.physical_qubits,
+                "xval": repetition,
+                "unit": "gate number",
+            }
+
+            circuits.append(circuit)
+
+        return circuits
+
+
+class FineXDrag(FineDrag):
+    """Class to fine calibrate the DRAG parameter of an X gate."""
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine amplitude experiment.
+
+        Experiment Options:
+            add_sx (bool): By default do not prepend an SX gate to the circuits.
+        """
+        options = super()._default_experiment_options()
+        options.add_sx = False
+
+        return options
+
+
+class FineSXDrag(FineDrag):
+    """Class to fine calibrate the DRAG parameter of an SX gate."""
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine amplitude experiment.
+
+        Experiment Options:
+            add_sx (bool): By default prepend an SX gate to the circuits.
+        """
+        options = super()._default_experiment_options()
+        options.add_sx = True
+
+        return options
+
+
+class FineDragEF(FineDrag):
```

Comment:
Maybe `FineEFDrag`? Or `FineDragX` and `FindDragSX` for others?

Reply:
I implicitly used the convention `Fine{pulse}Drag{transition}` this also aligns with fine amplitude so I would leave it as is for now.